artpaint/controls/NumberSliderControl.cpp

```diff
@@ -187,11 +187,12 @@ NumberSliderControl::_InitMessage()
 		fMessage = new BMessage;
 
 	if (fMessage) {
-		if (fMessage->HasInt32("value"))
+		if (!fMessage->HasInt32("value"))	// may have been set by creator
 			fMessage->AddInt32("value", 0);
 
-		if (fMessage->HasBool("final"))
-			fMessage->AddBool("final", true);
+//		if (fMessage->HasBool("final"))
+	// Should always have this -- often ignored:
```

Comment:
Maybe remove the commented line above and rather explain why "final" should always be true and what's it for.